tools/http_server.py

```diff
@@ -53,12 +53,14 @@ def spawn():
     r_thread.daemon = True
     r_thread.start()
     sleep(1)  # TODO I'm too lazy to figure out how to do this properly.
+    return thread
 
 
 if __name__ == '__main__':
     try:
-        spawn()
-        while True:
-            sleep(100)
+        thread = spawn()
+        while thread.is_alive():
+            sleep(10)
     except KeyboardInterrupt:
-        sys.exit()
+        pass
+    sys.exit(1)
```

Comment:
this has exit code 1 on sigint

Reply:
Is that not the correct behavior?

Reply:
I guess it's debatable, but I prefer 0

Reply:
I tried on several programs, most seem to use 130 actually... except python which uses 1. So perhaps the `pass` should be `sys.exit(130)`!